dokan_fuse/src/dokanfuse.cpp

```diff
@@ -433,6 +433,87 @@ static NTSTATUS DOKAN_CALLBACK FuseUnmounted(PDOKAN_FILE_INFO DokanFileInfo) {
   return errno_to_ntstatus_error(impl->unmounted(DokanFileInfo));
 }
 
+static DWORD AddSeSecurityNamePrivilege() {
+    LUID luid;
+    if (!LookupPrivilegeValue(0, SE_SECURITY_NAME, &luid)) {
+        DWORD err = GetLastError();
+        if (err != ERROR_SUCCESS) {
+            return err;
+        }
+    }
+
+    HANDLE token = 0;
+    if (!OpenProcessToken(GetCurrentProcess(),
+        TOKEN_ADJUST_PRIVILEGES | TOKEN_QUERY, &token)) {
+        return GetLastError();
+    }
+
+    ON_BLOCK_EXIT([token] {
+        CloseHandle(token);
+    });
+
+    LUID_AND_ATTRIBUTES attr;
+    attr.Attributes = SE_PRIVILEGE_ENABLED;
+    attr.Luid = luid;
+
+    TOKEN_PRIVILEGES priv;
+    priv.PrivilegeCount = 1;
+    priv.Privileges[0] = attr;
+
+    AdjustTokenPrivileges(token, FALSE, &priv, NULL, NULL, NULL);
+    DWORD err = GetLastError();
+    if (err != ERROR_SUCCESS) {
+        return err;
+    }
+
+    return ERROR_SUCCESS;
+}
+
+static NTSTATUS DOKAN_CALLBACK FuseGetFileSecurity(
+    LPCWSTR FileName, PSECURITY_INFORMATION SecurityInformation,
+    PSECURITY_DESCRIPTOR SecurityDescriptor, ULONG BufferLength,
+    PULONG LengthNeeded, PDOKAN_FILE_INFO DokanFileInfo) {
+  impl_fuse_context *impl = the_impl;
+  if (impl->debug())
+    FWPRINTF(stderr, L"GetFileSecurity %x\n", *SecurityInformation);
+
+  bool needASS = ((*SecurityInformation & SACL_SECURITY_INFORMATION) ||
+      (*SecurityInformation & BACKUP_SECURITY_INFORMATION));
+
+  if (needASS) {
+      DWORD result = AddSeSecurityNamePrivilege();
+      if (ERROR_SUCCESS != result)
+          return lasterror_to_ntstatus(result);
+  }
+
+  HANDLE handle = CreateFileW(
+      DokanFileInfo->DokanOptions->MountPoint,
```

Comment:
Hello @matepek ,

I was meaning that FuseGetFileSecurity is always going to return the same informations since it always open the same file when he should return the specific information related to the file requested.

Therefore, I am not sure the system will like to get informations of a directory when he request a file.
Maybe the implementation should return different values when it is a file, directory,...requested ?

I am also not really familiar with SECURITY_INFORMATION :cry: sorry.
https://msdn.microsoft.com/fr-fr/library/windows/desktop/aa379573%28v=vs.85%29.aspx